### PR TITLE
[inductor] overlap scheduling nodes benchmarking logs and fix

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #169947

1. Fix: benchmark_collective_with_cuda_events_impl was benchmarking without convertion FakeTensor -> real
2. Adding logging to tlparse of runtime estimations 'benchmarked' vs 'analytical', will be logged only if 'benchmark' mode is on. (In this case overhead for analytical calculation is minimal).
3. As we do warmup=1, nruns=3, using median for stability, as warmup 1 can be not enough. median is more stable for outliers in this case.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: torch/_inductor/fx_passes/overlap_scheduling.py [hunk 1]
```diff
@@ -498,6 +546,61 @@ def _calculate_compute_node_domination_index(self) -> dict[fx.Node, int]:
 
         return domination_index
 
+    def _log_compute_estimations(
```
Comment (eellison) [inline torch/_inductor/fx_passes/overlap_scheduling.py]: Can we put this code in node_runtime_estimation, just to try avoid this file getting too large?

File: torch/_inductor/fx_passes/overlap_scheduling.py [hunk 2]
```diff
@@ -506,6 +609,16 @@ def _log_collective_benchmarks(
         world_size: int,
     ) -> None:
         """Log collective benchmarks with analytical comparisons for tlparse."""
+        headers = [
```
Comment (eellison) [inline torch/_inductor/fx_passes/overlap_scheduling.py]: maybe we move this node runtime estimation as well..

File: torch/_inductor/fx_passes/node_runtime_estimation.py [hunk 3]
```diff
@@ -68,19 +74,33 @@ def _benchmark_collective_with_cuda_events_impl(
...
-    import torch.distributed as c10d
+    from torch._dynamo.testing import rand_strided
+
+    # Convert FakeTensors to real tensors before benchmarking
+    def to_real(t: torch.Tensor) -> torch.Tensor:
+        shape = [get_hint(dim) for dim in t.shape]
+        stride = [get_hint(s) for s in t.stride()]
+
+        if any(s is None for s in itertools.chain(shape, stride)):
+            raise ValueError("Cannot convert tensor with symbolic dimensions")
```
Comment (eellison) [inline torch/_inductor/fx_passes/node_runtime_estimation.py:87]: this shouldn't occur because can_benchmark_collective should have already checked unbacked.. maybe update error msg? / add comment

File: torch/_inductor/fx_passes/overlap_scheduling.py [hunk 4]
```diff
@@ -26,6 +28,52 @@
 from ..pattern_matcher import stable_topological_sort
 
 
+def _format_table_with_tabulate(
```
Comment (eellison) [inline torch/_inductor/fx_passes/overlap_scheduling.py]: similarly, are you sure we dont have a function for this in torch/_inductor/utils, of not , put it there ? `

**Review by eellison — APPROVED** [review]
Looks good ! one small comment on code movement

File: torch/_inductor/fx_passes/overlap_scheduling.py (hunk 4 above)
Comment (IvanKobzarev) [reply torch/_inductor/fx_passes/overlap_scheduling.py]: Oh, I did not know that tabulate_2d exist - I will replace the usage with tabulate_2d.

File: torch/_inductor/fx_passes/overlap_scheduling.py (hunk 1 above)
Comment (IvanKobzarev) [reply torch/_inductor/fx_passes/overlap_scheduling.py]: yes, will move all _log* functions there.
